gems/quilt_rails/README.md

```diff
@@ -326,67 +326,42 @@ function App() {
 export default App;
 ```
 
-##### Example: sending headers from Rails controller
+##### Example: Sending headers and custom data from Rails controller
+
+In some cases you may want to send custom headers or basic data from Rails to your React server. Quilt facilitates this case by providing consumers with a `data` argument on the `render_react` call.
+
+**Note:** The data passed should be data that is unlikely or will never change over the course of the session before they render any React components.
 
 ```ruby
 class ReactController < ApplicationController
   include Quilt::ReactRenderable
 
   def index
-    render_react(headers: { 'x-custom-header': 'header-value-a' })
+    render_react(data: { 'x-custom-header': 'header-value-a', 'some_id': 123 })
```

Comment:
I don't think `x-custom-header` should be passed into `data`, should it?

Reply:
No, it shouldn't. This is wrong. In this case, the `x-custom-header` header would be hidden behind the `x-quilt-data` header which is not what we'd want for a header.

Reply:
To support the separation of headers/data while ensuring it's easily accessible on the client, I've merged the headers and data in the `x-quilt-data` response header to the react-server.